### PR TITLE
Introduce shared HTTP permissions to avoid duplication across multiple HTTP permissions and to simplify creation of HTTP permissions

### LINKED ISSUE #37414: Enhancing Quarkus Security for ROOT Group Access
### Description

Assume you are using JWT for security which has a group called ROOT. The user with the group `ROOT` should be able to access all JAX-RS (all beans in general) despite what is defined in `@RolesAllowed`, `@PermissionsAllowed` or security config properties.

For example, the following configuration behavior does not allow that. The `root-policy` would work for all paths except the ones defined in `quarkus.http.auth.permission.*.paths` properties. In the following case the ROOT won't be able to access `/api/hello` due to the longest path wins principle `(/api/hello > /*)`.

```properties
quarkus.http.auth.policy.root-policy.roles-allowed=ROOT
quarkus.http.auth.permission.root.paths=/*
quarkus.http.auth.permission.root.policy=root-policy
quarkus.http.auth.policy.customer-policy.roles-allowed=CUSTOMER
quarkus.http.auth.permission.customer.paths=/api/hello
quarkus.http.auth.permission.customer.policy=customer-policy
```

The current behaviour does offer a solution which would mean that i will have to add the ROOT role to all `quarkus.http.auth.policy.*.roles-allowed` properties.

### Implementation ideas

A potential solution could involve incorporating a mechanism where, if a SecurityInterceptor is defined, it operates subsequent to the Quarkus Engine's determination of whether a call should be permitted or denied. In this proposed solution, the SecurityInterceptor would yield the conclusive result for the call after the Quarkus Engine has made its decision.

```java
interface SecurityInterceptor {
       Uni<CheckResult> isPermitted(CheckResult quarkusCheckResult);
}

@RequestScoped
class CustomSecurityInterceptor implements SecurityInterceptor {

       @Inject
       JsonWebToken jwt;

       Uni<CheckResult> isPermitted(Uni<CheckResult> quarkusCheckResult) {

           if(jwt.getGroups().contains("ROOT")){
               return Uni.createFrom().item(CheckResult.PERMIT);
           }

           return quarkusCheckResult;
       }
}
```

### PR DESCRIPTION
**Pull request by michalvavrik**
closes: #37414

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/PolicyMappingConfig.java [hunk 1]
```diff
@@ -60,4 +60,13 @@ public class PolicyMappingConfig {
      */
     @ConfigItem
     public Optional<String> authMechanism;
+
+    /**
+     * Shared permission checks are always applied whenever the path is matched. On the other hand, there is always
```
Comment (sberyozkin) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/PolicyMappingConfig.java:65]: @michalvavrik I suggest to simplify a bit here and expand if necessary in the docs, as it reads a little bit complex. How about something like `Indicates that this policy always applies to the matched paths in addition to the policy with a winning path. Avoid creating more than one shared policy to minimize the performance impact.` ?

**Review by sberyozkin — APPROVED** [review]
Thanks @michalvavrik, minor suggestion is proposed

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/PolicyMappingConfig.java (hunk 1 above)
Comment (michalvavrik) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/PolicyMappingConfig.java]: Sure, makes sense, fixed.

**Comment by sberyozkin** [comment]
Looks good to go

**Pull request merged**